docs/specification/draft/basic/authentication.md

```diff
@@ -0,0 +1,417 @@
+---
+title: Authentication
+type: docs
+weight: 15
+---
+
+{{< callout type="info" >}} **Protocol Revision**: draft {{< /callout >}}
+
+## 1. Introduction
+
+### 1.1 Purpose and Scope
+
+The Model Context Protocol provides authentication capabilities at the transport level,
+enabling MCP clients to make requests to restricted resources on behalf of resource
+owners. This specification defines the authentication flow for HTTP+SSE transport.
+
+### 1.2 Protocol Requirements
+
+MCP clients and servers supporting HTTP+SSE transport **SHOULD** implement this
+specification, though they **MAY** support additional authentication mechanisms.
+Implementations using alternative transports **MAY** implement authentication according
+to the best practices for their specific transport.
+
+For deployments supporting third-party clients or servers via HTTP+SSE, implementations
+**SHOULD** expect authentication as defined in this document. However, controlled
+environment deployments like corporate networks **MAY** implement alternative methods
+such as certificate-based authentication.
+
+### 1.3 Standards Compliance
+
+This authentication mechanism implements following specifications but recommends a
+specific subsets:
+
+- [OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12)
+- OAuth 2.0 Authentication Server Metadata
+  ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414))
+- OAuth 2.0 Dynamic Client Registration Protocol
+  ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591))
+
+## 2. Authentication Flow
+
+### 2.1 Overview
+
+MCP is designed to support interactive, user-facing developer tools that connect to
+arbitrary servers. This leads to two key design decisions:
+
+1. All MCP clients **MUST** be treated as public OAuth 2.1 clients, as they cannot
+   securely store client secrets when distributed to end users
+
+2. PKCE **MUST** be implemented by all clients to prevent authorization code interception
+   attacks, which is especially important for locally-running tools
```

Comment:
Refer to OAuth 2.1 here? (Notably it doesn't call it "PKCE" anymore, just incorporates it directly.)